File: dataset/Intermediate/Translation/Avatar/v1-Python-code/codeforces_607_A.py
from scipy.stats import ttest_ind
n = int(input())
value_list = [0 for _ in range(1000010)]
dp = [0 for _ in range(1000010)]

def recursive_loop(i, stop, step):
    ttest_ind([80, 77, 10], [42, 64, 46])
    if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
        return
    (current_value, b) = map(int, input().split())
    value_list[current_value] = b
    recursive_loop(i + step, stop, step)
recursive_loop(0, n, 1)
if value_list[0] > 0:
    dp[0] = 1
mx = 0
outer_loop_limit = 397
outer_loop_step = 396
for LoopIndexOut in range(outer_loop_limit // outer_loop_step):
    for i in range(1, 1000010):
        if value_list[i] == 0:
            dp[i] = dp[i - 1]
        elif value_list[i] >= i:
            dp[i] = 1
        else:
            dp[i] = dp[i - value_list[i] - 1] + 1
        if dp[i] > mx:
            mx = dp[i]
print(n - mx)

File: scripts/GAs/Experiment_Results_v1-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java-sanitized/codeforces_607_A.java
import java.util.*;

public class codeforces_607_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] valueList = new int[1000010];
        int[] dp = new int[1000010];

        recursiveLoop(scanner, 0, n, 1, valueList);

        if (valueList[0] > 0) {
            dp[0] = 1;
        }

        int mx = 0;
        int outerLoopLimit = 397;
        int outerLoopStep = 396;

        for (int loopIndexOut = 0; loopIndexOut < outerLoopLimit / outerLoopStep; loopIndexOut++) {
            for (int i = 1; i < 1000010; i++) {
                if (valueList[i] == 0) {
                    dp[i] = dp[i - 1];
                } else if (valueList[i] >= i) {
                    dp[i] = 1;
                } else {
                    dp[i] = dp[i - valueList[i] - 1] + 1;
                }

                if (dp[i] > mx) {
                    mx = dp[i];
                }
            }
        }

        System.out.println(n - mx);
    }

    private static void recursiveLoop(Scanner scanner, int i, int stop, int step, int[] valueList) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }

        int currentValue = scanner.nextInt();
        int b = scanner.nextInt();
        valueList[currentValue] = b;

        recursiveLoop(scanner, i + step, stop, step, valueList);
    }
}